app/src/main/java/cm/aptoide/pt/download/DownloadAnalytics.java

```diff
@@ -256,10 +257,19 @@ public void updateDownloadEvent(String versionCode, String packageName, int file
   }
 
   public void startProgress(Download download) {
-    cache.get(download.getPackageName() + download.getVersionCode() + DOWNLOAD_EVENT_NAME)
-        .setHadProgress(true);
-    cache.get(download.getMd5() + DOWNLOAD_COMPLETE_EVENT)
-        .setHadProgress(true);
+    startProgressOnDownloadEvent(
+        download.getPackageName() + download.getVersionCode() + DOWNLOAD_EVENT_NAME);
+    startProgressOnDownloadEvent(download.getMd5() + DOWNLOAD_COMPLETE_EVENT);
+  }
+
+  private void startProgressOnDownloadEvent(String key) {
```

Comment:
This event name is not clear without seeing what is inside. I suggest updateEventWithHasProgress or something along these lines.